edx_toggles/toggles/internal/waffle.py

```diff
@@ -0,0 +1,339 @@
+        name = self.waffle_namespace._namespaced_name(self.switch_name)
+        super().__init__(name, default=False, module_name=module_name)
+
+    @property
+    def namespaced_switch_name(self):
+        """
+        For backward compatibility, we still provide the `namespaced_switch_name`, property, even though users should
+        now use the `name` attribute.
+        """
+        return self.name
+
+    def is_enabled(self):
+        return self.waffle_namespace.is_enabled(self.switch_name)
+
+    @contextmanager
+    def override(self, active=True):
+        with self.waffle_namespace.override(self.switch_name, active):
+            yield
+
+
+class WaffleFlagNamespace(BaseNamespace):
+    """
+    Provides a single namespace for a set of waffle flags.
+
+    All namespaced flag values are stored in a single request cache containing
+    all flags for all namespaces.
+    """
+
+    @property
+    def _cached_flags(self):
+        """
+        Returns a dictionary of all namespaced flags in the request cache.
+        """
+        return _get_waffle_namespace_request_cache().setdefault("flags", {})
+
+    def is_flag_active(self, flag_name):
+        """
+        Returns and caches whether the provided flag is active.
+
+        If the flag value is already cached in the request, it is returned.
+
+        Note: A waffle flag's default is False if not defined. If you think you
+            need the default to be True, see the module docstring for
+            alternatives.
+
+        Arguments:
+            flag_name (String): The name of the flag to check.
+        """
+        namespaced_flag_name = self._namespaced_name(flag_name)
+        value = self._get_flag_active(namespaced_flag_name)
+        self._monitor_value(namespaced_flag_name, value)
+        return value
+
+    def _get_flag_active(self, namespaced_flag_name):
+        """
+        Return the value of the flag activation without touching the _waffle_flag_attribute.
+        """
+        # Check namespace cache
+        value = self._cached_flags.get(namespaced_flag_name)
+        if value is not None:
+            return value
+
+        # Check in context of request
+        request = crum.get_current_request()
+        value = self._get_flag_active_request(namespaced_flag_name, request)
+        if value is not None:
+            return value
+
+        # Return default value
+        return self._get_flag_active_default(namespaced_flag_name)
+
+    def _get_flag_active_request(self, namespaced_flag_name, request):
+        """
+        Get flag value in the context of the current request.
+        """
+        if request:
+            value = flag_is_active(request, namespaced_flag_name)
+            self._cached_flags[namespaced_flag_name] = value
+            return value
+        return None
+
+    def _get_flag_active_default(self, namespaced_flag_name):
+        """
+        Return default value in the absence of any other, more specific flag value.
```

Comment:
This name and docstring don't really match the implementation, which includes lots of warnings, because this is an exceptional case.  What about `_get_flag_active_for_missing_request` or `_get_flag_active_default_for_missing_request`?

Reply:
I'll go with `_get_flag_active_no_request` to be consisdent with the naming of the `_get_flag_active_request` method.